src/ngx_http_lua_worker_thread.c

```diff
@@ -0,0 +1,410 @@
+/*
+ * Copyright (C) Yichun Zhang (agentzh)
+ */
+
+
+#ifndef DDEBUG
+#define DDEBUG 0
+#endif
+#include "ddebug.h"
+
+
+#include "ngx_http_lua_worker_thread.h"
+#include "ngx_http_lua_util.h"
+
+
+#if (NGX_THREADS)
+
+
+#include <ngx_thread.h>
+#include <ngx_thread_pool.h>
+
+
+typedef struct task_ctx_s {
+    lua_State*          vm;
+    ngx_thread_mutex_t  mutex;
+    struct task_ctx_s*  next;
+} task_ctx_t;
+
+
+typedef struct {
+    task_ctx_t              *ctx;
+    ngx_http_lua_co_ctx_t   *wait_co_ctx;
+    ngx_http_cleanup_pt      cleanup;
+    void                    *cleanup_data;
+    int                      n_args;
+    int                      rc;
+    int                      is_abort;
+} my_thread_ctx_t;

```

Comment:
change to a more meaningful name ? my_thread_ctx_t   =>  ngx_http_lua_thread_ctx_t?